docs/role-based-access-control.rst

```diff
@@ -0,0 +1,188 @@
+created by pushing an image using ``podman`` or ``docker`` client. This same permission allows the user
+of Pulp's API to create a new namespace.
+
+The creation of a new namespace creates three user groups that can access the namespace:
+Owners, Collaborators, and Consumers. The user that creates the namespace is automatically added to
+the Owners group.
+
+Namespace Owners
+~~~~~~~~~~~~~~~~
+
+The group name is ``container.namespace.owners.<namespace name>``. This group has the following
+object permissions for the namespace::
+
+     "container.view_containernamespace"
+     "container.delete_containernamespace"
+     "container.namespace_add_containerdistribution",
+     "container.namespace_delete_containerdistribution
+     "container.namespace_view_containerdistribution"
+     "container.namespace_pull_containerdistribution"
+     "container.namespace_push_containerdistribution"
+     "container.namespace_change_containerdistribution"
+     "container.namespace_view_containerpushrepository"
+     "container.namespace_modify_content_containerpushrepository"
+
+The users in the owners group have the permissions to add/remove users from all three groups
+associated with the namespace. They also have the ability to create, update, and delete
+repositories in the namespace.
+
+In addition to being able to use the ``podman`` or ``docker`` client to manage repositories, owners
+can use Pulp's API to add and remove tags in the repositories for the namespace.
+
+Namespace Collaborators
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The group name is ``container.namespace.collaborators.<namespace name>``. This group has the
+following object permissions for the namespace::
+
+    "container.view_containernamespace"
+    "container.namespace_add_containerdistribution"
+    "container.namespace_delete_containerdistribution"
+    "container.namespace_view_containerdistribution"
+    "container.namespace_pull_containerdistribution"
+    "container.namespace_push_containerdistribution"
+    "container.namespace_change_containerdistribution"
+    "container.namespace_view_containerpushrepository"
+    "container.namespace_modify_content_containerpushrepository"
+
+Users in the Collaborator group can everything that the owners can, with the exception of deleting
+the namespace.
+
+Namespace Consumers
+~~~~~~~~~~~~~~~~~~~
+
+The group name is ``container.namespace.consumers.<namespace name>``. This group has the following
+object permissions for the namespace::
+
+    "container.view_containernamespace"
+    "container.namespace_view_containerdistribution"
+    "container.namespace_pull_containerdistribution"
+    "container.namespace_view_containerpushrepository"
+
+Users in the Consumers group can the ``pull`` any of the repositories in the namespace. Users should
+only need to be added to this group if private repositories are being used. If the repository is
+public, then anyone can ``pull`` from the repository.
```

Comment:
let me know if this is a bad idea, but we need to somehow separate and clarify the difference between pulp repo and registry repo which in pulp world translates to distribution+pulp-repo

If we use 'registry repository' instead of just 'repository' will it be any better?

Reply:
on the other hand after finishing reading all the docs it is pretty much clear, user just need to be patient and read it all :smiley: